### DISCUSSION TITLE
How to "Enable Realtime" when using API (anon)?

### DISCUSSION THREAD

**Discussion by evt3135** [supabase/supabase#12726] [Questions] [answered]
I am able to "Enable Realtime" by executing the below create_table() function in the SQL Editor. However, when I invoke this create_table() function from an API call, it errors out saying:
"must be owner of publication supabase_realtime."

If I change the owner of table from anon to postgres, it errors out saying:
"must be member of role postgres"

Please advice how can I Enable Realtime from an API call. Thank you in advance.

EVT

Here is the function:

```
CREATE OR REPLACE FUNCTION create_table(input_table text)

RETURNS void AS 

$$

BEGIN

EXECUTE
format('CREATE TABLE %I AS TABLE table_default;

ALTER TABLE %I OWNER TO postgres;

ALTER PUBLICATION supabase_realtime ADD TABLE %I;',input_table, input_table, input_table);

END;

$$ 

LANGUAGE 'plpgsql';
```

**Comment by mansueli** [comment, marked as answer]
Please set the function as SECURITY DEFINER:

https://www.postgresql.org/docs/current/sql-createfunction.html

I'll also mark this as answered as you already told me in support that it worked 🫡
